### PR TITLE
feat: Add support for MySQL database, fixes RHOAIENG-3246, RHOAIENG-3247

### PR DESCRIPTION
**Pull request by dhirajsb**
## How Has This Been Tested?

Unit test updated to test both DB configs. 
Tested locally by using the updated mysql and postgres samples.

## Merge criteria:

- [X] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dhirajsb** [comment]
MySQL test is timing out. Will take a look in the morning.

**Comment by dhirajsb** [comment]
For some reason the mysql DB in the ci Kind cluster fails to start with the error:
```
mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)'
Check that mysqld is running and that the socket: '/var/run/mysqld/mysqld.sock' exists!
```

But the same mysql 8.3.0 image runs fine in a local Kind cluster and a ROSA OpenShift cluster. 
I've switched the test back to using postgres so that this Kind test for CI can pass. It still exercises the operator template changes so it's a valid test.

File: api/v1alpha1/modelregistry_webhook.go [hunk 1]
```diff
@@ -61,6 +62,14 @@ func (r *ModelRegistry) Default() {
 	if len(r.Spec.Rest.Image) == 0 {
 		r.Spec.Rest.Image = config.GetStringConfigWithDefault(config.RestImage, config.DefaultRestImage)
 	}
+
+	// FIX default database configs that get set for some reason in Kind cluster
```
Comment (lampajr) [inline api/v1alpha1/modelregistry_webhook.go:66]: Is this something would you like to fix in this PR, I am not that sure I got the issue here.

File: api/v1alpha1/modelregistry_types.go [hunk 2]
```diff
@@ -135,10 +188,15 @@ type ModelRegistrySpec struct {
 	// Configuration for gRPC endpoint
 	Grpc GrpcSpec `json:"grpc"`
 
-	//+kubebuilder:required
+	// One of Postgres or MySQL database connections must be specified!
```
Comment (tarilabs) [inline api/v1alpha1/modelregistry_types.go:191]: is there a way we can maybe add this as part of RestSpec comment, so to see it in resulting docs?

in other words my question is: if this comment is left here, will it be present in resulting generated docs, or maybe best to place this comment on top of line `type RestSpec struct {` 137 so to have it included?

File: api/v1alpha1/modelregistry_webhook.go (hunk 1 above)
Comment (dhirajsb) [reply api/v1alpha1/modelregistry_webhook.go]: Changed comment to Fixes...

**Review by lampajr — APPROVED** [review]
lgtm

**Pull request merged**